factory/core/coreComponentsHandler.go

```diff
@@ -372,6 +372,18 @@ func (mcc *managedCoreComponents) GenesisTime() time.Time {
 	return mcc.coreComponents.genesisTime
 }
 
+// SupernovaGenesisTime returns the time for supernova round activation
+func (mcc *managedCoreComponents) SupernovaGenesisTime() time.Time {
+	mcc.mutCoreComponents.RLock()
+	defer mcc.mutCoreComponents.RUnlock()
+
+	if mcc.coreComponents == nil {
+		return time.Time{}
+	}
+
+	return mcc.coreComponents.supernovaGenesisTime
```

Comment:
one refactoring idea is to move everything related to time in a component (like chronology)

Reply:
right now, we have genesis time in core components here, but also in genesis nodes config handler